datafusion/core/tests/sql/select.rs

```diff
@@ -106,9 +105,9 @@ async fn test_prepare_statement() -> Result<()> {
     let ctx = create_ctx_with_partition(&tmp_dir, partition_count).await?;
 
     // sql to statement then to prepare logical plan with parameters
-    // c1 defined as UINT32, c2 defined as UInt64 but the params are Int32 and Float64
-    let dataframe =
-        ctx.sql("PREPARE my_plan(INT, DOUBLE) AS SELECT c1, c2 FROM test WHERE c1 > $2 AND c1 < $1").await?;
```

Comment:
PREPARE now returns an empty DataFrame and can't call `with_param_values` on it. I think this change is reasonable because the result of `PREPARE` is not a real relation, as it doesn't contain valid rows.

Reply:
I agree -- PREPARE is a statement in my mind (and has no results)